src/udev/udev-builtin-net_id.c

```diff
@@ -600,6 +603,101 @@ static int names_platform(sd_device *dev, NetNames *names, bool test) {
         return 0;
 }
 
+#define DEVICETREE_ALIASES "/proc/device-tree/aliases"
+
+static int dev_devicetree_onboard(sd_device *dev, NetNames *names) {
+        char *ofnode;
+        const char *parent_syspath;
+        _cleanup_free_ char *devicetree_syspath = NULL;
+        _cleanup_free_ char *ofnode_syspath = NULL;
+        _cleanup_free_ char *ofnode_path = NULL;
+        _cleanup_closedir_ DIR *aliases = NULL;
+        sd_device *parent = NULL;
+        int r;
+        size_t ofnode_pathlen;
+        unsigned int index = 0;
+
+        if (!naming_scheme_has(NAMING_DEVICETREE_ALIASES))
+                return 0;
+
+        /* check if our direct parent has an of_node */
+        r = sd_device_get_parent(dev, &parent);
+        if (r < 0)
+                return r;
+
+        r = sd_device_get_syspath(parent, &parent_syspath);
+        if (r < 0)
+                return r;
+
+        ofnode = strjoina(parent_syspath, "/of_node");
+        r = chase_symlinks(ofnode, NULL, 0, &ofnode_syspath, NULL);
+        if (r < 0)
+                return r;
```

Comment:
Maybe, you can use `sd_device_new_child()` and `sd_device_get_syspath()` for the child device?

Reply:
Sorry, what is the child device here? This is a symlink to something in the devicetree firmware, outside of `/sys/devices`. Is it OK to do that? We really just want to get the relative path, which is what the alias will reference.

Reply:
Not familiar with device tree. But if the resolved path (/sys/firmware/devicetree/base/soc/ethernet@deadbeef in the example below) is a directory, then new_child() should work fine.

Reply:
OK, I wasn't sure whether the `sd_device_*` stuff was for just `/sys/devices`, or for anything in `/sys`.

Reply:
You can easily test by calling `udevadm info /sys/firmware/devicetree/base/soc/ethernet@deadbeef`.

Reply:
Thanks, that was very helpful.